Fix: Increase login_attempt.ip_address column length for IPv6 support

Jguer opened the pull request:
- Expand ip_address column from VARCHAR(30) to VARCHAR(50) to accommodate IPv6 addresses
- Add database migration with support for PostgreSQL, MySQL, and SQLite
- Add comprehensive integration tests for various IPv6 address formats
- Resolves 500 errors when login fails over IPv6, now returns proper 401 errors

Fixes #106362

Jguer committed:
fix: increase login_attempt.ip_address column length for IPv6 support

- Expand ip_address column from VARCHAR(30) to VARCHAR(50) to accommodate IPv6 addresses
- Add database migration with support for PostgreSQL, MySQL, and SQLite
- Add comprehensive integration tests for various IPv6 address formats
- Resolves 500 errors when login fails over IPv6, now returns proper 401 errors

Fixes #106362

Jguer committed:
test: add missing test skip to TestIntegrationIPv6AddressSupport

Skip integration test when running with -short flag to separate unit and integration tests

Jguer committed:
Update pkg/services/sqlstore/migrations/login_attempt_mig.go

Co-authored-by: Victor Cinaglia <victor@grafana.com>

Jguer committed:
fix missing bracket

cinaglia commented on pkg/services/loginattempt/loginattemptimpl/login_attempt_test.go:
```diff
@@ -205,6 +205,58 @@ func TestIPLoginAttempts(t *testing.T) {
+			assert.NoError(t, err, "Should be able to add login attempt with IPv6 address: %s", ipAddress)
+
+			// Verify that the login attempt was stored correctly
+			count, err := service.store.GetIPLoginAttemptCount(ctx, GetIPLoginAttemptCountQuery{IPAddress: ipAddress})
```
It looks like you might also need to provide the `Since` property when defining `GetIPLoginAttemptCountQuery`.

Jguer committed:
fix: resolve PostgreSQL timestamp overflow in IPv6 test

- Use controlled time mock instead of time.Now() to avoid timestamp conversion issues
- Follow existing test patterns with xormStore and mock time functions
- Add proper Since parameter to GetIPLoginAttemptCount query
- Fixes PostgreSQL error: 'pq: value "-62135596800" is out of range for type integer'

Jguer committed:
fix: resolve PostgreSQL UTF-8 encoding error in IPv6 test

Replace string(rune(i)) with fmt.Sprintf to avoid null bytes (0x00)
when i=0, which caused 'invalid byte sequence for encoding UTF8' error

cinaglia reviewed (approved):
LGTM
